src/main/java/formflow/library/FileController.java

```diff
@@ -111,6 +120,10 @@ public ResponseEntity<?> upload(
         }
       }
 
+      if (userFileRepositoryService.countBySubmission(submission) >= maxFiles) {
```

Comment:
If we had 20 files all uploaded at the same time (through distinct but parallel requests), this might not be enough logic to prevent more than the max from being uploaded, given the test and set conundrum (first you test then you set, but there's a gap of time between testing and setting). But good enough for now.